vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -435,8 +484,15 @@ def request_finished(
             params["do_remote_prefill"] = False
             return False, None
 
-        if (not params.get("do_remote_decode")
-                or request.status != RequestStatus.FINISHED_LENGTH_CAPPED):
+        if not params.get("do_remote_decode"):
+            return False, None
+
+        if request.status != RequestStatus.FINISHED_LENGTH_CAPPED:
+            if request.request_id in self._reqs_need_send:
```

Comment:
Corner case - a request is aborted after it has been finished? Can this happen? We don't have a unit test that covers it